python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -681,10 +708,18 @@ def _pack_value(
             _pack_value(item, whitelist_map, f"{descent_path}[{idx}]")
             for idx, item in enumerate(cast(list, val))
         ]
+    if tval is AssetKeyMap:
```

Comment:
nit: you can double check in the benchmark but moving this if check down since the condition is more rare may be measurably beneficial